heartbeat/LVM-activate

```diff
@@ -0,0 +1,830 @@
+#!/bin/bash
+#
+#
+# Copyright (c) 2017 SUSE LINUX, Eric Ren
+#			All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it would be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# Further, this software is distributed without any warranty that it is
+# free of the rightful claim of any third person regarding infringement
+# or the like.	Any license provided herein, whether implied or
+# otherwise, applies only to this software file.  Patent licenses, if
+# any, provided herein do not apply to combinations of this program with
+# other software, or any other product whatsoever.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write the Free Software Foundation,
+# Inc., 59 Temple Place - Suite 330, Boston MA 02111-1307, USA.
+#
+# LVM-activate OCF Resource Agent:
+#
+# Logical volume manager (LVM) provides new features for cluster enviroment:
+# lvmlockd and system ID, which aims to replace clvmd and tagged-exclusive
+# activation. Accordingly, we have created a new resource agent named "lvmlockd"
+# to manage lvmlockd daemon. In addition, this new resource agent "LVM-activate"
+# is created to take care of LVM activation/deactivation work. This agent supports
+# the new features: lvmlockd and system ID, and also supports the old features:
+# clvmd and lvm tag.
+#
+# For LVM activation, we already have an old resource agent "LVM" that supports
+# LVM1, clvmd and lvm tag. Some reasons why we tend to create a new one rather
+# than changing the old:
+# 1. "LVM" supports clvmd and some legacy things. The code is already rather complex,
+# which make it difficult to change and ensure everything works correctly.
+#
+# 2. We want this new RA to do the activation work in a much more safe way by
+# leveraging the new features and activating at VG or LV basis depending on
+# your choice.
```

Comment:
But couldn't there still be a way to update the existing RA? The problem with having two RAs: a) maintenance effort, b) introduction of issues which were already fixed due to extensive field testing, c) users confused even more than necessary (i.e. which RA should I use?). The two points above don't seem to me to be arguments strong and clear enough for the introduction of a new RA.

Reply:
Good point. We (the redhat developers and me) initially discussed this concern. We think it would be almost complete rewrite of the old LVM RA if we want to support the new features and add some more parameters for the RA. So we decided to create a new one to see how much difficult to merge them.

Yes, it's time to discuss if we should keep this new one or merge it with the old one. My concerns are:
1. Is it acceptable to change the parameters of the old LVM?

2. Are there some principle rules to follow when I start to merge such a huge change to the old LVM RA?

Reply:
maintenance effort:  the new RA is much simpler to maintain, and the old RA should not require  further maintenance.

introduction of issues: this is a reason to leave the old RA alone while the new one is phased in.  The vast majority of the junk in the old RA has no value and should be removed, not legitimate fixes.

users confused: changing the parameters on the same RA would be the most confusing it seems to me.

which RA should I use: the new one.